prov/verbs/src/verbs_domain_xrc.c

```diff
@@ -183,13 +188,22 @@ void fi_ibv_put_shared_ini_conn(struct fi_ibv_xrc_ep *ep)
 	if (ep->base_ep.id)
 		ep->base_ep.id->qp = NULL;
 
+	/* If XRC physical QP connection was not completed, make sure
+	 * any pending connection to that destination will get scheduled. */
+	if (ep->base_ep.id && ep->base_ep.id == ini_conn->phys_conn_id) {
+		if (ini_conn->state == FI_IBV_INI_QP_CONNECTING)
+			ini_conn->state = FI_IBV_INI_QP_UNCONNECTED;
+		ini_conn->phys_conn_id = NULL;
```

Comment:
A debug level message would be nice here. We've migrated a connection from connecting state to unconnected.

Reply:
Ok, but this is expected behavior that is common and we see it on every simultaneous connection handled by RXM. I'll add a debug log. Thanks.

Reply:
In that case, let's omit it. It looked like a strange transition

Reply:
Sounds good, thanks.